public/assets/js/scripts.js

```diff
@@ -100,225 +118,246 @@ jQuery(document).ready(function() {
 	});
 
 	// doctor step
-	$('#doctor_step').on('click', function() {
+	$('#doctor_step').on('click', function () {
 		var data = $('#doctor').serializeArray();
 
 		if (!verifyAllFilled(data)){
 			alert('Please fill out all the fields correctly before proceeding.');
 		}
 
-		else{
+		if (!verifyAllFilled(data)) {
+			alert('Please fill out all the fields correctly before proceeding.');
+		} else {
 			doctor_json.name = data[0].value;
-	    	doctor_json.designation = data[1].value;
-	    	doctor_json.registration_number = data[2].value;
-	    	doctor_json.location = data[3].value;
+			doctor_json.designation = data[1].value;
+			doctor_json.registration_number = data[2].value;
+			doctor_json.location = data[3].value;
 
-	    	loadNextStep('#doctor');
-	    }
+			loadNextStep('#doctor');
+		}
 	});
 
-
 	// patient step
-	$('#patient_step').on('click', function() {
+	$('#patient_step').on('click', function () {
 		patient_json = new Object();
```

Comment:
`patient_json` is being reset here. You initialised with some empty fields already above. So you can remove this.